proposals/lip-introduce_validators_module.md

```diff
@@ -0,0 +1,577 @@
+
+
+To mitigate this risk, we propose to add an extra key pair to a validator account that is used for creating block signatures. Then, a generator node only requires access to the secret generator key, but not to the secret that is used for signing transactions. To increase security even further, validators should be able to update the generator key pair at any time. This is done by calling the `setValidatorGeneratorKey` function, described [below](#setValidatorGeneratorKey).
+
+
+### Ensuring Uniqueness of validator BLS Keys
+
+The validators module store maintains an account for each validator registered in the chain. In particular, it stores the [BLS key](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0038.md) associated with the validator, used to sign commits. BLS keys have to be unique across the chain, i.e. two validators are not allowed to register the same BLS key. To easily check whether a BLS key has been previously registered, we use a _registered BLS keys substore_, to store all the registered validator BLS keys. Store keys are set to the BLS key and the corresponding store value to the address of the validator that registered the key. This allows to check for the existence of a certain BLS key in constant time.
+
+### Block Slots
+
+In the Lisk protocol, time is divided into intervals of fixed length. 
+These intervals are called a _block slots_ and at most one block can be added to the blockchain during each slot. 
+The length of a block slot is a constant called _block time_.
+The validator module provides the functions to convert timestamp to block slots and viceversa.
+Furthermore, it maintains the _generator list_, an array of addresses corresponding to the accounts that are eligible to generate a block in the current round. 
+Each entry of the generator list corresponds to a block slot.
+
+## Specification
+
+In this section, we specify the substores that are part of the validators module store and the protocol logic called during the block processing. The validators module has module ID `MODULE_ID_VALIDATORS` (see the [table below](#notation-and-constants)).
+
+
+### Notation and Constants
+
+We define the following constants:
+
+
+| Name          | Type    | Value       | Description |
+| ------------- |---------| ------------| ------------|
+|`MODULE_ID_VALIDATORS` | uint32 | TBD | ID of the validators module.|
+|`STORE_PREFIX_VALIDATORS_DATA` | bytes | 0x0000 | Store prefix of the validators data substore.|
+|`STORE_PREFIX_GENERATOR_LIST` | bytes | 0x4000 | Store prefix of the generator list substore.|
+|`STORE_PREFIX_BLS_KEYS` | bytes | 0x8000 | Store prefix of the registered BLS keys substore.|
+|`STORE_PREFIX_GENESIS_DATA` | bytes | 0xc000 | Store prefix of the genesis data substore.|
+|`INVALID_BLS_KEY` | bytes | 48 bytes all set to 0x00 | An invalid BLS key, used as a placeholder before a valid BLS key is registered.|
+|`BLOCK_TIME` | integer | Block time (in seconds) set in the chain configuration. |
```

Comment:
the description of this one is placed in the value column. My suggestion for value is to set `10 (value for the Lisk mainchain)`

Reply:
done